tensorly/tenalg/proximal.py

```diff
@@ -593,17 +593,25 @@ def simplex_prox(tensor, parameter):
             "Validation of subgradient optimization."
             Mathematical programming 6.1 (1974): 62-88.
     """
-    _, col = tl.shape(tensor)
-    tensor = tl.clip(tensor, 0, tl.max(tensor))
+    # Making it work for 1-dimensional tensors as well
+    if tl.ndim(tensor) > 1:
+        row, col = tl.shape(tensor)
+    else:
+        row = tl.shape(tensor)[0]
+        col = 1
+        tensor = tl.reshape(tensor, [row, col])
     tensor_sort = tl.sort(tensor, axis=0, descending=True)
-
-    to_change = tl.sum(tl.where(tensor_sort > (tl.cumsum(tensor_sort, axis=0) - parameter), 1.0, 0.0), axis=0)
+    # Broadcasting is used to divide rows by 1,2,3...
+    cumsum_min_param_by_k = (tl.cumsum(tensor_sort, axis=0) - parameter) / tl.cumsum(tl.ones([row, 1]), axis=0)
+    # Added -1 to correspond to a Python index
+    to_change = tl.sum(tl.where(tensor_sort > cumsum_min_param_by_k, 1.0, 0.0), axis=0) - 1
     difference = tl.zeros(col)
     for i in range(col):
-        if to_change[i] > 0:
-            difference = tl.index_update(difference, tl.index[i], tl.cumsum(tensor_sort, axis=0)[int(to_change[i] - 1), i])
-    difference = (difference - parameter) / to_change
-    return tl.clip(tensor - difference, a_min=0)
+        difference = tl.index_update(difference, tl.index[i], cumsum_min_param_by_k[int(to_change[i]), i])
```

Comment:
the explicit casting to int of `to_change[i]` is causing issues in numpy -- does it not work if we don't explicitly cast?

Reply:
I removed casting and used `1` and `0` instead of `1.0` and `0.0` when creating `to_change`. However mxnet returns an error when testing not simplex but constrained parafac with simplex interestingly. I am trying to find a solution but if you have any idea, it would be nice to hear.

Reply:
What's the issue with MXNet?

Reply:
Now, there is no issue with any backend.